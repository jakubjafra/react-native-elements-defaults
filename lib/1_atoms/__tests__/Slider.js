import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

const utils = require("../../utils");

describe('with no theme applied', () => {
    let OldSlider, Slider;

    beforeAll(() => {
        let defaultStyles = {
            'atoms/slider': {
            }
        };

        OldSlider = require('react-native-elements/src/slider/Slider').default;

        Slider = utils.connectTheme(true, 'atoms/slider', require('../Slider'))(defaultStyles);
    });

    test('OldSlider and Slider should be different', () => {
        expect(OldSlider).not.toBe(Slider);
    });

    test('OldSlider renders correctly', () => {
        const tree = renderer.create(<OldSlider />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('Slider renders correctly', () => {
        const tree = renderer.create(<Slider />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('OldSlider and Slider returns the same thing if no theme is applied', () => {
        const oldTree = renderer.create(<OldSlider />).toJSON();
        const tree = renderer.create(<Slider />).toJSON();

        expect(JSON.stringify(oldTree)).toEqual(JSON.stringify(tree));
    });
});

describe('with basic theme applied', () => {
    let OldSlider, Slider;

    beforeAll(() => {
        let defaultStyles = {
            'atoms/slider': {
                trackStyle: {
                    backgroundColor: 'red',
                }
            }
        };

        OldSlider = require('react-native-elements/src/slider/Slider').default;

        Slider = utils.connectTheme(true, 'atoms/slider', require('../Slider'))(defaultStyles);
    });

    test('OldSlider and Slider should be different', () => {
        expect(OldSlider).not.toBe(Slider);
    });

    test('OldSlider renders correctly', () => {
        const tree = renderer.create(<OldSlider />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('Slider renders correctly', () => {
        const tree = renderer.create(<Slider />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('OldSlider and Slider should be different caused by theme applied', () => {
        const oldTree = renderer.create(<OldSlider />).toJSON();
        const tree = renderer.create(<Slider />).toJSON();

        expect(oldTree).not.toEqual(tree);
    });
});
