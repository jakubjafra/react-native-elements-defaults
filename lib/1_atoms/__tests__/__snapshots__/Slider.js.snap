// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with basic theme applied OldSlider renders correctly 1`] = `
<View
  animationType="timing"
  onLayout={[Function]}
  step={0}
  style={
    Array [
      Object {
        "height": 40,
        "justifyContent": "center",
      },
      false,
      undefined,
    ]
  }
  thumbTouchSize={
    Object {
      "height": 40,
      "width": 40,
    }
  }
  value={0}
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#b3b3b3",
        },
        Object {
          "borderRadius": 2,
          "height": 4,
        },
        undefined,
      ]
    }
  />
  <View
    collapsable={undefined}
    style={
      Object {
        "backgroundColor": "#3f3f3f",
        "borderRadius": 2,
        "height": 4,
        "marginTop": -0,
        "opacity": 0,
        "position": "absolute",
        "width": 0,
      }
    }
  />
  <View
    collapsable={undefined}
    onLayout={[Function]}
    style={
      Object {
        "backgroundColor": "red",
        "borderRadius": 10,
        "height": 20,
        "opacity": 0,
        "position": "absolute",
        "top": 22,
        "transform": Array [
          Object {
            "translateX": 0,
          },
          Object {
            "translateY": -0,
          },
        ],
        "width": 20,
      }
    }
  />
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {},
      ]
    }
  />
</View>
`;

exports[`with basic theme applied Slider renders correctly 1`] = `
<View
  animationType="timing"
  onLayout={[Function]}
  step={0}
  style={
    Array [
      Object {
        "height": 40,
        "justifyContent": "center",
      },
      false,
      undefined,
    ]
  }
  thumbTouchSize={
    Object {
      "height": 40,
      "width": 40,
    }
  }
  value={0}
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#b3b3b3",
        },
        Object {
          "borderRadius": 2,
          "height": 4,
        },
        Array [
          undefined,
          Object {
            "backgroundColor": "red",
          },
        ],
      ]
    }
  />
  <View
    collapsable={undefined}
    style={
      Object {
        "backgroundColor": "#3f3f3f",
        "borderRadius": 2,
        "height": 4,
        "marginTop": -0,
        "opacity": 0,
        "position": "absolute",
        "width": 0,
      }
    }
  />
  <View
    collapsable={undefined}
    onLayout={[Function]}
    style={
      Object {
        "backgroundColor": "red",
        "borderRadius": 10,
        "height": 20,
        "opacity": 0,
        "position": "absolute",
        "top": 22,
        "transform": Array [
          Object {
            "translateX": 0,
          },
          Object {
            "translateY": -0,
          },
        ],
        "width": 20,
      }
    }
  />
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {},
      ]
    }
  />
</View>
`;

exports[`with no theme applied OldSlider renders correctly 1`] = `
<View
  animationType="timing"
  onLayout={[Function]}
  step={0}
  style={
    Array [
      Object {
        "height": 40,
        "justifyContent": "center",
      },
      false,
      undefined,
    ]
  }
  thumbTouchSize={
    Object {
      "height": 40,
      "width": 40,
    }
  }
  value={0}
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#b3b3b3",
        },
        Object {
          "borderRadius": 2,
          "height": 4,
        },
        undefined,
      ]
    }
  />
  <View
    collapsable={undefined}
    style={
      Object {
        "backgroundColor": "#3f3f3f",
        "borderRadius": 2,
        "height": 4,
        "marginTop": -0,
        "opacity": 0,
        "position": "absolute",
        "width": 0,
      }
    }
  />
  <View
    collapsable={undefined}
    onLayout={[Function]}
    style={
      Object {
        "backgroundColor": "red",
        "borderRadius": 10,
        "height": 20,
        "opacity": 0,
        "position": "absolute",
        "top": 22,
        "transform": Array [
          Object {
            "translateX": 0,
          },
          Object {
            "translateY": -0,
          },
        ],
        "width": 20,
      }
    }
  />
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {},
      ]
    }
  />
</View>
`;

exports[`with no theme applied Slider renders correctly 1`] = `
<View
  animationType="timing"
  onLayout={[Function]}
  step={0}
  style={
    Array [
      Object {
        "height": 40,
        "justifyContent": "center",
      },
      false,
      undefined,
    ]
  }
  thumbTouchSize={
    Object {
      "height": 40,
      "width": 40,
    }
  }
  value={0}
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#b3b3b3",
        },
        Object {
          "borderRadius": 2,
          "height": 4,
        },
        undefined,
      ]
    }
  />
  <View
    collapsable={undefined}
    style={
      Object {
        "backgroundColor": "#3f3f3f",
        "borderRadius": 2,
        "height": 4,
        "marginTop": -0,
        "opacity": 0,
        "position": "absolute",
        "width": 0,
      }
    }
  />
  <View
    collapsable={undefined}
    onLayout={[Function]}
    style={
      Object {
        "backgroundColor": "red",
        "borderRadius": 10,
        "height": 20,
        "opacity": 0,
        "position": "absolute",
        "top": 22,
        "transform": Array [
          Object {
            "translateX": 0,
          },
          Object {
            "translateY": -0,
          },
        ],
        "width": 20,
      }
    }
  />
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {},
      ]
    }
  />
</View>
`;
